pyClient/test_commands/scenario.py

```diff
@@ -76,24 +77,26 @@ def bob_deposit(
         (output_note1, pk_bob),
         (output_note2, pk_bob)])
 
-    # Hash the pk_sender and cipher-texts
-    ciphers = pk_sender + ciphertexts[0] + ciphertexts[1]
-    hash_ciphers = sha256(ciphers).hexdigest()
+    # Hashing all inputs of the signature
+    # Encode the ciphertexts and ephemeral encryption key
+    data_to_be_signed = pk_sender + ciphertexts[0] + ciphertexts[1]
 
-    # Hash the proof
+    # Encode the proof
     proof: List[str] = []
     for key in proof_json.keys():
         if key != "inputs":
             proof.extend(proof_json[key])
-    hash_proof = sha256(encode_to_hash(proof)).hexdigest()
+    data_to_be_signed += encode_to_hash(proof)
 
-    # Encode and hash the primary inputs
-    encoded_inputs = joinsplit.encode_pub_input_to_hash(proof_json["inputs"])
-    hash_inputs = sha256(encoded_inputs).hexdigest()
+    # Encode the primary inputs
+    encoded_inputs = encode_to_hash(proof_json["inputs"])
+    data_to_be_signed += encoded_inputs
+
+    # Hash data_to_be_sign
+    hash_tobesign = sha256(data_to_be_signed).hexdigest()
```

Comment:
Why don't use do sha256(pk || c0 || c1 || proof || inputs)? instead of calling `encode_to_hash` multiple times @rrtoledo ?

Reply:
I do not directly use sha because we need to encode in bytes everything.

I also want to make sure of the order we hash these arguments for the verifying process on the contract side.

Reply:
Also, encode_to_hash does not take bytes as inputs.